### PR TITLE
Apply coupons automatically

### PR DESCRIPTION
**Pull request by ryanmitchell**
@sampoyigi forgot to open a PR for this. Do you think it needs any more than this?

### CODE REVIEW

**Comment by sampoyigi** [comment]
Yup looks like you've got it all covered. I'll come back for another review ;)

File: models/Coupons_model.php [hunk 1]
```diff
@@ -103,6 +104,11 @@ public function scopeIsEnabled($query)
         return $query->where('status', '1');
     }
 
+    public function scopeIsAutomatic($query)
```
Comment (sampoyigi) [inline models/Coupons_model.php]: Change to `scopeIsAutoApplicable` or something descriptive enough

File: database/migrations/2020_11_01_000300_create_apply_automatically.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Igniter\Coupons\Database\Migrations;
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateApplyAutomatically extends Migration
```
Comment (sampoyigi) [inline database/migrations/2020_11_01_000300_create_apply_automatically.php]: Rename class to `AddAutoApplyFieldOnCouponsTable`, change the file name to match too.

File: database/migrations/2020_11_01_000300_create_apply_automatically.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public function up()
+    {
+        Schema::table('igniter_coupons', function (Blueprint $table) {
+            $table->boolean('apply_automatically')->default(FALSE);
```
Comment (sampoyigi) [inline database/migrations/2020_11_01_000300_create_apply_automatically.php]: rename column to `auto_apply`

**Review by sampoyigi — CHANGES_REQUESTED** [review]

File: models/config/coupons_model.php [hunk 4]
```diff
@@ -174,13 +174,21 @@
...
             'default' => 'forever',
+            'span' => 'left',
             'options' => [
                 'forever' => 'lang:igniter.coupons::default.text_forever',
                 'fixed' => 'lang:igniter.coupons::default.text_fixed',
                 'period' => 'lang:igniter.coupons::default.text_period',
                 'recurring' => 'lang:igniter.coupons::default.text_recurring',
             ],
         ],
+        'apply_automatically' => [
```
Comment (sampoyigi) [inline models/config/coupons_model.php]: you missed this, also check the coupons_model $casts attributes and maybe rename the lang string to match?

Please do another test to confirm all works as expected so I can merge immediately after ;)

**Review by sampoyigi — CHANGES_REQUESTED** [review]

File: models/config/coupons_model.php (hunk 4 above)
Comment (ryanmitchell) [reply models/config/coupons_model.php]: Changed and tested.

**Review by sampoyigi — APPROVED** [review]

**Pull request merged**